cmd/traefik/traefik.go

```diff
@@ -119,6 +120,11 @@ func runCmd(staticConfiguration *static.Configuration) error {
 	}
 
 	ctx, _ := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	ctx, svcCancel := context.WithCancel(ctx)
+	go func() {
+		<-winsvc.ChanExit
```

Comment:
At first glance on other platforms, this goroutine will never end, as the channel will never receive a message. I am wondering if this mechanism could be encapsulated to only deal with ctx in the traefik cmd.